inttest/sonobuoy/terraform/k0sctl.tf

```diff
@@ -0,0 +1,40 @@
+
+data "http" "k0s_version" {
+  count = local.use_remote_k0s_version ? 1 : 0
+  url   = "https://docs.k0sproject.io/${var.k0s_version}.txt"
+}
+
+locals {
+  use_remote_k0s_version = var.k0s_version == "stable" || var.k0s_version == "latest"
+  k0sctl_tmpl = {
+    apiVersion = "k0sctl.k0sproject.io/v1beta1"
+    kind       = "cluster"
+    metadata = {
+      name = local.cluster_unique_identifier
+    }
+    spec = {
+      hosts = [
+        for host in concat(aws_eip.controller-ext, aws_instance.cluster-workers) : {
+          ssh = {
+            address = host.public_ip
+            user    = "ubuntu"
+            keyPath = "./aws_private.pem"
+          }
+          role          = host.tags["Role"]
+          uploadBinary  = true
+          k0sBinaryPath = local.use_remote_k0s_version ? null : var.k0s_binary_path
+        }
+      ]
+      k0s = {
+        version = local.use_remote_k0s_version ? chomp(data.http.k0s_version.0.body) : var.k0s_version
+      }
+    }
+  }
+}
+
+// Save the private key to filesystem
+resource "local_file" "k0sctl_config" {
+  file_permission = "600"
+  filename        = format("%s/%s", path.module, "k0sctl.yaml")
+  content         = replace(yamlencode(local.k0sctl_tmpl), "/((?:^|\n)[\\s-]*)\"([\\w-]+)\":/", "$1$2:")
```

Comment:
This feels overly complex, why do we need to do such replaces?

On most other cases with TF & k0sctl, we do something like this:
```
output "k0s_cluster" {
    value = yamlencode(local.k0s_tmpl)
}
```
which we can feed into k0sctl:
```
terraform output -raw k0s_cluster | k0sctl apply --config -
```
IMO this is bit cleaner than storing the (unneeded) temp k0sctl.yaml file and all the possible replaces etc. like above

Reply:
I feel the file-based approach is better for when needing to run the conformance test manually. it's easier to see what k0sctl is actually doing. The regex replace here is used to output a better looking yaml file, since terrafor's yamlencode outputs a yaml that has all values and keys quoted. I felt the output after the regex replacement looks more like our standard yaml configuration files (see https://github.com/hashicorp/terraform/issues/23322).

Reply:
Given that the regex is rather a cosmetic thing, I'm not sure if it's carrying its own weight. It's already on the verge of being intimidating :smiling_face_with_tear: